### ISSUE TITLE
Tripplr - LBNP - compliance check

### ISSUE THREAD

**Issue by abhinavv245** [ONDC-Official/v1.2.0-logs#2661]
Multiple issues identified. Please use the [log verification tool](https://github.com/ONDC-Official/reference-implementations/tree/main/utilities/logistics-b2b/log-verification-utility) and fix all the identified issues and resubmit the logs.
Also submit, ready_to_ship at /confirm flow and buyer cancellation flow for TAT Breach.

@punithrj26

**Comment by punithrj26** [comment]
Hi @abhinavv245,

I have used log verification tool before uploading this log.
The errors got are from logistics seller app, to which we don't have control on. We are working on logistic buyer app.

We have uploaded three flow as mentioned,
1) defining catalog.
2) Order fulfilment.
3) Rto flow.

Kindly let me know whether this is correct or not. Any changes needed in above mentioned flows. Thank you.

**Comment by abhinavv245** [comment]
@punithrj26 There are issues from buyer side as well. Please check for all the timestamps. In case there are schematic issues from LSP side, please handle at your end. 

Also submit, ready_to_ship at /confirm flow and buyer cancellation flow for TAT Breach, as per this [sheet](https://docs.google.com/spreadsheets/d/1JZV6ZQzXcHUsOwegGtArX3DdIXYIy3gxkhQ00q7kICc/edit?gid=1670900093#gid=1670900093) (Group B)

**Comment by punithrj26** [comment]
Hi @abhinavv245 ,

When we run log verification for order fulfilment, we are getting few error related to lsp,

For example,
1. "on_init_0": {
            "quotePriceErr": "Quote price 49 does not match the breakup total  49.09 in on_init"
        }
2. "on_status_1": {
            "trckErr": "tracking should be enabled (true) for hyperlocal (Immediate Delivery)"
        }
3.  "on_track_0": {
            "orderIdErr": "Order Id in /track and /on_track does not match"
        }

In the earlier conversation you had mentioned that if there are schematic issues from LSP side, please handle at your end.

Does this mean, I need to send "NACK" for the schema issue or should I make changes in the logs.
Can you be more specific what needs to be done in this kind of scenario.

Thanks.

**Comment by abhinavv245** [comment]
You need to NACK in such cases. Also, I see some of the issues are due to
the context timestamps. Please see those are as per the flow.

**Comment by abhinavv245** [comment]
@punithrj26 Have you implemented deferred RTS(ready to ship) flow as well? (/update and /on_update calls)

**Comment by punithrj26** [comment]
@abhinavv245 
Yes we have implemented rts cancellation flow have added in logs.

**Comment by abhinavv245** [comment]
Can  you provide any link as I cannot see deferred RTS flow. (/update call)

@punithrj26

**Comment by punithrj26** [comment]
@abhinavv245 
Here it is https://github.com/ONDC-Official/v1.2.0-logs/tree/master/Tipplr/Tipplr%20Logistics%20Buyer/tipplr-LBNP-logs(logistics)-22-OCT/Buyer%20Cancel%20Flow

**Comment by abhinavv245** [comment]
@punithrj26 I cannot see /update and /on_update calls.

**Comment by punithrj26** [comment]
@abhinavv245 
Sorry my bad have missed to add these in this flow.

Can you please give feedback on other two flows.

**Comment by punithrj26** [comment]
@abhinavv245 
Can you please provide feedback on remaining flows.

**Comment by abhinavv245** [comment]
## Cancel Flow

- ready to ship is marked in confirm, why is update call needed?

- update call is required when the order is marked ready to ship after confirmation of the order.

@punithrj26

**Comment by punithrj26** [comment]
@abhinavv245 
Please refer to this folder, confirm file: https://github.com/ONDC-Official/v1.2.0-logs/blob/master/Tipplr/Tipplr%20Logistics%20Buyer/tipplr-LBNP-logs(logistics)-22-OCT/Cancel%20Flow-RTS/confirm.json

rts not given at confirm call.

**Comment by abhinavv245** [comment]
Logs approved for v1.2.

@punithrj26

**Comment by punithrj26** [comment]
Hi @abhinavv245,
Thank you.

Should we upload IGM and RSF logs and get it verified as well before going in to production ?
